chain/client/src/sync.rs

```diff
@@ -52,15 +52,8 @@ pub const MAX_PENDING_PART: u64 = MAX_STATE_PART_REQUEST * 10000;
 pub const NS_PER_SECOND: u128 = 1_000_000_000;
 
 /// Get random peer from the hightest height peers.
-pub fn highest_height_peer(highest_height_peers: &Vec<FullPeerInfo>) -> Option<FullPeerInfo> {
-    if highest_height_peers.len() == 0 {
-        return None;
-    }
-
-    match highest_height_peers.iter().choose(&mut thread_rng()) {
-        None => highest_height_peers.choose(&mut thread_rng()).cloned(),
-        Some(peer) => Some(peer.clone()),
-    }
+pub fn highest_height_peer(highest_height_peers: &[FullPeerInfo]) -> Option<FullPeerInfo> {
```

Comment:
It may be a good idea to simply inline this method. I think that the inline code is easier to read. By in-lining the code, the reader will not have to check what the method does, as it should be obvious,

I would rather see:
```
         if let Some(peer) = highest_height_peers.choose(&mut thread_rng()).cloned() {
```
than
```
         if let Some(peer) = highest_height_peer(highest_height_peers) {
```

Reply:
I looked at the code. It looks like the clone may be be even needed in some of the cases.

Reply:
good point, didn't even think of that...